datafusion/src/sql/planner.rs

```diff
@@ -200,7 +211,9 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
     /// Generate a logic plan from an SQL query
     pub fn query_to_plan(&self, query: &Query) -> Result<LogicalPlan> {
-        self.query_to_plan_with_alias(query, None, &mut HashMap::new())
+        self.safe_recursion(|| {
```

Comment:
Determine if it is safe to call this function without stack overflow, if it's not safe, return `Err(Recursion exceeded limit)`